apps/api/src/controllers/languages.controller.test.ts

```
@@ -0,0 +1,565 @@
+    const allLangsWithSpecialChars = [langSpecialChars1, langSpecialChars2, langSpecialChars3];
+
+    beforeEach(() => {
+      // Default mock for db.select().from().where().orderBy().limit().offset() to return empty array
+      // and db.select({ count: count() }).from().where() to return [{ count: 0 }]
+      // This can be overridden in specific tests if needed.
+
+      const mockQueryChain = {
+        orderBy: mock().mockReturnThis(),
+        limit: mock().mockReturnThis(),
+        offset: mock().mockReturnThis(),
+        then: (callback: any) => callback([]), // Default to empty array for results
```

Biome (1.9.4)

[error] Do not add then to an object.



(lint/suspicious/noThenProperty)